microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/corrections/proposal/OpenAPIAnnotationProposal.java

```diff
@@ -0,0 +1,116 @@
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copied from /org.eclipse.jdt.ui/src/org/eclipse/jdt/internal/ui/text/correction/proposals/NewAnnotationMemberProposal.java
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.redhat.microprofile.jdt.core.java.corrections.proposal;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.Annotation;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.IExtendedModifier;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.core.dom.MemberValuePair;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.NormalAnnotation;
+import org.eclipse.jdt.core.dom.StringLiteral;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite.ImportRewriteContext;
+import org.eclipse.jdt.internal.corext.codemanipulation.ContextSensitiveImportRewriteContext;
+import org.eclipse.lsp4j.CodeActionKind;
+
+
+public class OpenAPIAnnotationProposal extends ASTRewriteCorrectionProposal {
+
+	private final CompilationUnit fInvocationNode;
+	private final TypeDeclaration fTypeNode;
+	private final String fAnnotation;
+	
+	public OpenAPIAnnotationProposal(String label, ICompilationUnit targetCU, CompilationUnit invocationNode, 
+			TypeDeclaration type, String annotation, int relevance) {
+		super(label, CodeActionKind.Source, targetCU, null, relevance);
+		this.fInvocationNode = invocationNode;
+		this.fTypeNode = type;
+		this.fAnnotation = annotation;
+	}
+	
+	@Override
+	protected ASTRewrite getRewrite() throws CoreException {
+		MethodDeclaration methods[] = this.fTypeNode.getMethods();
+		List<MethodDeclaration> responseReturnMethods = new ArrayList<>();
+		for (MethodDeclaration method : methods) {
+			boolean operationFlag = false;
+			if (method.getReturnType2().resolveBinding().getQualifiedName().equals("javax.ws.rs.core.Response")) {
+				List<?> modifiers = method.modifiers();
+				for (Iterator<?> iter = modifiers.iterator(); iter.hasNext();) {
+					Object next = iter.next();
+					if (next instanceof IExtendedModifier) {
+						IExtendedModifier modifier = (IExtendedModifier) next;
+						if (modifier.isAnnotation()) {
+							Annotation annotation = (Annotation) modifier;
+							if (annotation.resolveTypeBinding().getQualifiedName()
+									.equals("org.eclipse.microprofile.openapi.annotations.Operation")) {
```

Comment:
Use your OPERATION_ANNOTATION  constants

Reply:
The constant is defined in a "internal" package, should I move it some where else?

Reply:
> The constant is defined in a "internal" package, should I move it some where else?

Yes the idea is to set those constants in non internal package, because other plugin could use this contsntas (ex : ConfigProperty for MP config

But OpenAPIAnnotationProposal should be hosted in a internal openapi package.